components/bl-multiple-select-component/src/options/index.js

```diff
@@ -23,25 +38,25 @@ export function Options({ type, options, selectValue, setSelectValue, onChange }
       newSelectValue = selectValue.filter(({ objectId }) => objectId !== isOptionSelected.objectId);
     } else {
       const selectedItems = [...selectValue, option];
-      
+
       newSelectValue = type === 'default'
         ? options.filter(item => selectedItems.includes(item))
```

Comment:
Here is N^2 complexity. You could create a map of options/selected items and check for existence by looking for the key in the map.